sasdata/dataloader/filereader.py

```diff
@@ -138,21 +149,23 @@ def nextline(self) -> str:
         Returns the next line in the file as a string.
         """
         #return self.f_open.readline()
-        return decode(self.f_open.readline())
+        return decode(self.string_open.readline())
 
     def nextlines(self) -> str:
         """
         Returns the next line in the file as a string.
         """
-        for line in self.f_open:
+        for line in self.string_open:
             #yield line
             yield decode(line)
 
     def readall(self) -> str:
         """
         Returns the entire file as a string.
         """
-        return decode(self.f_open.read())
+        handle = self.string_open if self.string_open else self.f_open
+        handle.seek(0)
+        return decode(handle.read())
```

Comment:
If `read` is called directly instead of `readall` then `seek` will not be triggered and the file may be in the wrong position. So go back to my original position: have the registry reset the seek position between reader attempts.